### ISSUE TITLE
uritemplate version incompatibilities

### ISSUE THREAD

**Issue by andreaTP** [microsoft/kiota-abstractions-python#63]
Noticed by installing a generated SDK in a project with more dependencies I get this error message:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behavior is the source of the following dependency conflicts.
google-api-python-client 1.12.11 requires uritemplate<4dev,>=3.0.0, but you have uritemplate 4.1.1 which is incompatible.
kfp 1.8.19 requires uritemplate<4,>=3.0.1, but you have uritemplate 4.1.1 which is incompatible.
```

Given the number of issues we are facing with `uritemplate` libraries across implementations, I'm wondering if it makes sense to roll our own for kiota to be more independent ... cc. @baywet

**Comment by baywet** [comment]
Thanks for reporting this @andreaTP as always.
I'm fairly novice to Python and currently on my phone.
Are those dependencies that are transiently pulling Uri template dependencies of the kiota packages or of your project?
Is there such a thing as a peer (npm) or internal dependency (implementation with gradle) with pip?

**Comment by andreaTP** [comment]
@baywet sorry for the delay:

> Are those dependencies that are transiently pulling Uri template dependencies of the kiota packages or of your project?

They are transitive dependencies within a Data Science Jupyter notebook, they come from "my project", or better, from my environment.

> Is there such a thing as a peer (npm) or internal dependency (implementation with gradle) with pip?

I quickly looked up this, but I cannot find anything relevant (not a Python deps expert too though).

**Comment by baywet** [comment]
@samwelkanda could you provide more context on dependency management for Python here please?

**Comment by andreaTP** [comment]
Closed by #118
